delivery_carrier_service_level/models/stock_picking.py

```diff
@@ -10,14 +10,20 @@ class StockPicking(models.Model):
     _name = "stock.picking"
     _inherit = "stock.picking"
 
+    def _compute_carrier_service_level(self):
+        carrier_model = self.env["carrier.service.level"]
+        for this in self:
+            this.carrier_service_level_id = False
+            if not this.carrier_id:
+                continue
+            this.carrier_service_level_id = carrier_model.search(
```

Comment:
```
this.carrier_service_level_id = service_level_model,search(
    [("carrier_id", "=", this.carrier_id.id)],
    limit=1,
) if this.carrier_id else service_level_model.browse([])
```
So assign empty record_set to this.carrier_service_level_id if there is no carrier.